pkg/api/controller.go

```diff
@@ -3006,6 +3011,63 @@ func (c *Controller) GetCurrentUser(w http.ResponseWriter, r *http.Request) {
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) sendResetPasswordEmail(email string, token string) error {
+	err := c.Emailer.SendEmail([]string{email}, token, token, nil)
+	if err != nil {
+		c.Logger.WithError(err).WithField("email", email).Debug("failed sending reset password email")
+		return err
+	}
+	c.Logger.WithField("email", email).Info("Successfully sent reset password email")
+	return nil
+}
+
+func (c *Controller) PasswordForgot(w http.ResponseWriter, r *http.Request, body PasswordForgotJSONRequestBody) {
+	user, err := c.Auth.GetUserByEmail(r.Context(), body.Email)
+	if err != nil {
+		c.Logger.WithError(err).WithField("request_body", body).Debug("failed to retrieve user by email")
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+	secret := c.Auth.SecretStore().SharedSecret()
+	currentTime := time.Now()
+	token, err := GenerateJWTResetPassword(secret, user.ID, body.Email, currentTime, currentTime.Add(tokenExpiryDuration))
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err)
+		c.Logger.WithError(err).WithField("email", *user.Email).Debug("failed to create a token")
+		return
+	}
+	// TODO (@shimi9276) create template for sending the email with link for reset
+	err = c.sendResetPasswordEmail(*user.Email, token)
+	if err != nil {
+		c.Logger.WithError(err).WithField("email", user.Email).Debug("failed sending reset password email")
+		return
+	}
+	c.Logger.WithField("email", user.Email).Info("Successfully sent reset password email")
+	writeResponse(w, http.StatusNoContent, err)
+}
+
+func (c *Controller) UpdatePassword(w http.ResponseWriter, r *http.Request, body UpdatePasswordJSONRequestBody) {
+	claims, err := VerifyResetPasswordToken(c.Auth, body.Token)
+	if err != nil {
+		writeError(w, http.StatusUnauthorized, err)
+		c.Logger.WithError(err).WithField("token", body.Token).Debug("failed to verify token")
```

Comment:
previous feedback.  log and write response.

Reply:
Fixed in all places (as above)